File: win-prog/ProgWinCS/Dialog Boxes/ModelessColorScroll/ModelessColorScroll.cs
//--------------------------------------------------
// ModelessColorScroll.cs © 2001 by Charles Petzold
//--------------------------------------------------
using System;
using System.Drawing;
using System.Windows.Forms;

class ModelessColorScroll: Form
{
     public static void Main()
     {
          Application.Run(new ModelessColorScroll());
     }
     public ModelessColorScroll()
     {
          Text = "Modeless Color Scroll";

          ColorScrollDialogBox dlg = new ColorScrollDialogBox();
          
          dlg.Owner = this;
          dlg.Color = BackColor;
          dlg.Changed += new EventHandler(ColorScrollOnChanged);
          dlg.Show();
     }
     void ColorScrollOnChanged(object obj, EventArgs ea)
     {
          ColorScrollDialogBox dlg = (ColorScrollDialogBox) obj;

          BackColor = dlg.Color;
     }
}

File: win-prog/ProgWinVB/Dialog Boxes/ModelessColorScroll/ModelessColorScroll.vb
'----------------------------------------------------
' ModelessColorScroll.vb (c) 2002 by Charles Petzold
'----------------------------------------------------
Imports System
Imports System.Drawing
Imports System.Windows.Forms

Class ModelessColorScroll
    Inherits Form

    Shared Sub Main()
        Application.Run(New ModelessColorScroll())
    End Sub

    Sub New()
        Text = "Modeless Color Scroll"

        Dim dlg As New ColorScrollDialogBox()
        dlg.Owner = Me
        dlg.Color = BackColor
        AddHandler dlg.Changed, AddressOf ColorScrollOnChanged
        dlg.Show()
    End Sub

    Sub ColorScrollOnChanged(ByVal obj As Object, ByVal ea As EventArgs)
        Dim dlg As ColorScrollDialogBox = _
                            DirectCast(obj, ColorScrollDialogBox)
        BackColor = dlg.Color
    End Sub
End Class
